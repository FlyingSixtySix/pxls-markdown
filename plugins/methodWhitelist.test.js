const { processor, expectProcessorParse } = require('../testUtil.js');
const methodWhitelist = require('./methodWhitelist');

processor().use(methodWhitelist);

test('does not do anything when no options are passed', () => expectProcessorParse('# All _off_ **these** `should`\n1. still!\n2. work!').toMatchSnapshot());

test('does not tokenize "*not italic*"', () => {
	const configuredProcessor = processor().use(methodWhitelist, { inline: [] });
	expectProcessorParse('*not italic*', false, configuredProcessor);
});

test('does not tokenize "1. raw\\n2. list"', () => {
	const configuredProcessor = processor().use(methodWhitelist, { block: [] });
	expectProcessorParse('1. raw\n2. list', false, configuredProcessor);
});

