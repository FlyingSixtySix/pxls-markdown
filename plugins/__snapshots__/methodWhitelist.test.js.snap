// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not do anything when no options are passed 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "type": "text",
        "value": "All ",
      },
      Object {
        "children": Array [
          Object {
            "type": "text",
            "value": "off",
          },
        ],
        "type": "emphasis",
      },
      Object {
        "type": "text",
        "value": " ",
      },
      Object {
        "children": Array [
          Object {
            "type": "text",
            "value": "these",
          },
        ],
        "type": "strong",
      },
      Object {
        "type": "text",
        "value": " ",
      },
      Object {
        "type": "inlineCode",
        "value": "should",
      },
    ],
    "depth": 1,
    "type": "heading",
  },
  Object {
    "children": Array [
      Object {
        "checked": null,
        "children": Array [
          Object {
            "children": Array [
              Object {
                "type": "text",
                "value": "still!",
              },
            ],
            "type": "paragraph",
          },
        ],
        "spread": false,
        "type": "listItem",
      },
      Object {
        "checked": null,
        "children": Array [
          Object {
            "children": Array [
              Object {
                "type": "text",
                "value": "work!",
              },
            ],
            "type": "paragraph",
          },
        ],
        "spread": false,
        "type": "listItem",
      },
    ],
    "ordered": true,
    "spread": false,
    "start": 1,
    "type": "list",
  },
]
`;
